nav/virtual_layers/include/virtual_layers/virtual_layer.h

```diff
@@ -0,0 +1,47 @@
+#ifndef VIRTUAL_LAYER_H_
+#define VIRTUAL_LAYER_H_
+#include <ros/ros.h>
+#include <costmap_2d/layer.h>
+#include <costmap_2d/layered_costmap.h>
+#include <costmap_2d/GenericPluginConfig.h>
+#include <dynamic_reconfigure/server.h>
+#include <vector> 
+
+
+namespace virtual_layers
+{
+
+class VirtualLayer : public costmap_2d::Layer, public costmap_2d::Costmap2D
+{
+public:
+  VirtualLayer();
+
+  virtual void onInitialize();
+  virtual void updateBounds(double robot_x, double robot_y, double robot_yaw, double* min_x, double* min_y, double* max_x,
+                             double* max_y);
+  virtual void updateCosts(costmap_2d::Costmap2D& master_grid, int min_i, int min_j, int max_i, int max_j);
+  bool isDiscretized()
+  {
+    return true;
+  }
+  bool initialize = false;
+  std::vector<std::vector<double>> points;
+  
+  virtual void matchSize();
+
+  double counter = 0.0;
+
+  //void resetLastUpdated();
+
+private:
+  const double COSTMAP_OFFSET_X = 4.0; //25.0; //4.0;
```

Comment:
You should leave a comment here explaining that 4.0 is the constant for local and 25.0 for global. Then remove the comments on the right. 

Alternatively, you can define 4 constants (e.g. `LOCAL_COSTMAP_OFFSET_X`, `LOCAL_COSTMAP_OFFSET_Y`, `GLOBAL_COSTMAP_OFFSET_X`, `GLOBAL_COSTMAP_OFFSET_Y`) with the respective values. 

I'm leaning towards the first soln because it makes it faster to change code should we need to